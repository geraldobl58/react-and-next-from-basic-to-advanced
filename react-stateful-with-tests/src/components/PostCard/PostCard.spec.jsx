import { render, screen } from '@testing-library/react';
import { PostCard } from '.';
import { postCardPropsMock } from './mock';

const props = postCardPropsMock;

describe('<PostCard />', () => {
  it('should render PostCard correctly', () => {
    render(<PostCard {...props} />);

    expect(screen.getByRole('img', { name:  /Title 1/i }))
      .toHaveAttribute('src', 'img/img.png');

    expect(screen.getByRole('heading', { name: /Title 1/i }))
      .toBeInTheDocument();

    expect(screen.getByText('Body 1'))
      .toBeInTheDocument();

    expect(screen.getByAltText(/Title 1/i))
      .toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { container } = render(<PostCard {...props} />);

    expect(container.firstChild).toMatchSnapshot();
  })
}); 