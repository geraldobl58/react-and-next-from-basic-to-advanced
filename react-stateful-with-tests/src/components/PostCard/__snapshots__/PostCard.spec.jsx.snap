// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should match snapshot 1`] = `
<div
  class="post"
>
  <img
    alt="Title 1"
    src="img/img.png"
  />
  <div
    class="post-content"
  >
    <h3>
      Title 1
    </h3>
    <p>
      Body 1
    </p>
  </div>
</div>
`;
